lib/experimental/fonts-api/bc-layer/class-gutenberg-fonts-api-bc-layer.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Fonts API BC Layer helpers.
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ * This file is now part of the API's Backwards-Compatibility (BC) layer.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts API
+ * @since      15.7.0
+ */
+
+/**
+ * Class Gutenberg_Fonts_API_BC_Layer
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ */
+class Gutenberg_Fonts_API_BC_Layer {
+
+	/**
+	 * Determines if the given fonts array is the deprecated array structure.
+	 *
+	 * @param array $fonts Array of fonts to check.
+	 * @return bool True when deprecated structure, else false.
+	 */
+	public static function is_deprecated_structure( array $fonts ) {
+		// Checks if a first dimension key is empty or non-string.
+		foreach ( array_keys( $fonts ) as $font_family ) {
```

Comment:
Small suggestion to update the comment since now this code checks all the array keys.
Also, I suggest to get rid of `array_keys()`.
But this is just a nitpicking.
```suggestion
		// Checks if first dimension keys are empty or non-string.
		foreach ( $fonts as $font_family => $font ) {
```

Reply:
Good catch on the inline comment change 👍 

This method only needs the keys, not the values. While the `array_keys()` does introduce a performance hit, it's clear of the iteration's intent and avoids introducing an unused variable, i.e. `$font`. As this is in the BC Layer, performance is not a major concern as extenders should be changing their plugins to use the new array structure. In doing so, they also gain a tiny micro-optimization.

Are you okay with leaving it in @anton-vlasenko?

Reply:
Yes, I'm fine with leaving it in, @hellofromtonya. Definitely, it shouldn't be a performance hit.

Reply:
@anton-vlasenko I further improved the inline comment https://github.com/WordPress/gutenberg/pull/50077/commits/1207bc309baef76d342115da989a0b260a419048. What do you think? Better?

Reply:
Yes, it's better (not to say that is wasn't good before). Thanks.

Reply:
Awesome! Thanks @anton-vlasenko